vite.config.ts

```diff
@@ -1,14 +1,95 @@
+    //   buffer: 'rollup-plugin-node-polyfills/polyfills/buffer-es6',
+    //   process: 'rollup-plugin-node-polyfills/polyfills/process-es6',
+    //   // Not included in `rollup-plugin-node-polyfills`.
+    //   // os: 'rollup-plugin-node-polyfills/polyfills/os',
+    //   os: 'rollup-plugin-node-polyfills/polyfills/os',
+    //   // os: 'rollup-plugin-polyfill-node',
+    //   // sys: 'util',
+    //   // punycode: 'rollup-plugin-node-polyfills/polyfills/punycode',
+    //   // https: 'rollup-plugin-node-polyfills/polyfills/http',
+    //   // assert: 'rollup-plugin-node-polyfills/polyfills/assert',
+    //   // constants: 'rollup-plugin-node-polyfills/polyfills/constants',
+    //   // _stream_duplex:
+    //   //           'rollup-plugin-node-polyfills/polyfills/readable-stream/duplex',
+    //   // _stream_passthrough:
+    //   //           'rollup-plugin-node-polyfills/polyfills/readable-stream/passthrough',
+    //   // _stream_readable:
+    //   //           'rollup-plugin-node-polyfills/polyfills/readable-stream/readable',
+    //   // _stream_writable:
+    //   //           'rollup-plugin-node-polyfills/polyfills/readable-stream/writable',
+    //   // _stream_transform:
+    //   //           'rollup-plugin-node-polyfills/polyfills/readable-stream/transform',
+    //   // timers: 'rollup-plugin-node-polyfills/polyfills/timers',
+    //   // console: 'rollup-plugin-node-polyfills/polyfills/console',
+    //   // vm: 'rollup-plugin-node-polyfills/polyfills/vm',
+    //   // zlib: 'rollup-plugin-node-polyfills/polyfills/zlib',
+    //   // tty: 'rollup-plugin-node-polyfills/polyfills/tty',
+    //   // domain: 'rollup-plugin-node-polyfills/polyfills/domain',
+    // },
+    alias: {
+      util: 'rollup-plugin-polyfill-node',
+      events: 'rollup-plugin-polyfill-node',
+      stream: 'rollup-plugin-polyfill-node',
+      // path: 'rollup-plugin-polyfill-node', // 'extname' is not exported by `path`
+      path: 'rollup-plugin-node-polyfills/polyfills/path', // 'posix' is not exported by
+      querystring: 'rollup-plugin-polyfill-node',
+      url: 'rollup-plugin-polyfill-node',
+      string_decoder: 'rollup-plugin-polyfill-node',
+      http: 'rollup-plugin-polyfill-node',
+      buffer: 'rollup-plugin-polyfill-node',
+      process: 'rollup-plugin-polyfill-node',
+      os: 'rollup-plugin-polyfill-node'
+    },
+  },
+  optimizeDeps: {
+    esbuildOptions: {
+      // Node.js global to browser globalThis
+      define: {
+        global: 'globalThis'
+      },
+      // Enable esbuild polyfill plugins
+      plugins: [
+        NodeGlobalsPolyfillPlugin({
+          process: true,
+          buffer: true
+        }),
+        NodeModulesPolyfillPlugin()
+      ]
+    }
+  },
   build: {
     rollupOptions: {
       plugins: [
-        nodePolyfills(),
-      ],
-    },
-  },
+        // Enable rollup polyfills plugin
+        // used during production bundling
+        rollupNodePolyFill(),
```

Comment:
what are we using polyfills for?

Reply:
AWS SDK uses node.js built in modules. E.g. https://github.com/aws/aws-sdk-js-v3/blob/main/packages/credential-providers/src/fromTokenFile.ts#L5 imports another package that depends on `fs` and I believe there are a few other ones.

Using Node.js built-in and globals in a browser project is discouraged but sometimes there is a need to use some dependency package that makes some use of them. To let these packages work on the browser we should add some polyfill that helps mimic these missing packages. With Vite we need to specify those manually while some another bundlers automatically add polyfills.